### ISSUE TITLE
add --force flag to svc deploy to force a deployment

### ISSUE THREAD

**Issue by adamjkeller** [aws/copilot-cli#1400]
Problem: 

I made changes to my dockerfile locally, but don't necessarily want to commit to git yet. Running a copilot svc deploy returns an error:

```
✘ Failed to deploy service.

✘ deploy service: change set with name ecscli-4ad9db69-4d94-4daf-a8af-8a5b9696c0cc for stack ecsworkshop-test-ecsdemo-nodejs has no changes
```

Solution:

Add a --force (or --force-deployment to match aws cli) flag to trigger a force update to the service. 

```
copilot svc deploy [--force] [--force-deployment]
```

For reference:
aws cli:
https://docs.aws.amazon.com/cli/latest/reference/ecs/update-service.html (--force-new-deployment flag)

Console:
![Screen Shot 2020-09-18 at 2 59 51 PM](https://user-images.githubusercontent.com/7571023/93648471-b7721c00-f9bf-11ea-9394-9161f46a9635.png)

Documentation on updating a service in ECS:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/update-service.html

**Comment by efekarakus** [comment]
I think using `copilot svc deploy --tag someRandomTag` might allow you to mitigate this issue, but yeah some force flag would be more obvious

**Comment by jnnkky** [comment]
> I think using `copilot svc deploy --tag someRandomTag` might allow you to mitigate this issue, but yeah some force flag would be more obvious

This causes the same failure. Changing the tag doesnt do anything for me.

**Comment by ryanvade** [comment]
I've run into this issue as well though adding a tag seems to have fixed it. Perhaps since this is using commit sha as tag I should commit before deploying?

**Comment by iamhopaul123** [comment]
Yeah currently Copilot requires you to commit your changes of the repo if you want to deploy any code changes. Under the hood it is detecting the latest commit and changing the image tag.

So if you are building from the local Dockerfile then commit your change and do the redeployment should work. Or like @efekarakus said you could change the tag as well. However, if you are using the existing image instead of building from the local Dockerfile, your service/job won't get updated if your image URL remains the same (see #1621).

**Comment by noahjahn** [comment]
> So if you are building from the local Dockerfile then commit your change and do the redeployment should work.

I've committed and pushed changes to the GitHub repo for the project using Copilot deployment to ECS. I continue to receive same deployment failure message when trying use the same tag name `copilot deploy --env staging --name cms --tag staging`:
```
✘ Failed to deploy service.

✘ deploy service: change set with name copilot-redacted for stack redacted has no changes
```

My use case is pushing with the `staging` tag. Ideally, each deployment could just update that image with the same tag name so the images don't clutter the repository.

👍 to adding the force flag..

**Comment by efekarakus** [comment]
Hi folks! We released the fix for this issue in v1.6.0: https://github.com/aws/copilot-cli/releases/tag/v1.6.0 🥳 part of #2139

Now you can run `copilot deploy` with local changes and copilot will pick up the changes. Resolving the issue.
